### PR TITLE
Update color constants to use HEX codes

### LINKED ISSUE #59215: Standard color name returns non-standard color code.
### Godot version

3.4.3 stable

### System information

it does not matter

### Issue description

Standard color name (`brown`) returns non-standard color code. The standard brown color code is `#a52a2a` `Color.brown.to_html(false)` gives `#a62929`.  I just tried the brown. I haven't checked other color constants. I'm sure they are also problematic.

## 
```gdscript
extends Node

func _ready():
	var c := Color.brown.to_html(false) # -> a62929
	# Standard X11/HTML Brown -------------> a52a2a - rgb(165, 42, 42)
	print(c)
	#a62929
```

## Godot Documentation
https://docs.godotengine.org/en/latest/classes/class_color.html#color
> You can also create a color from standardized color names by using the string constructor or directly using the color constants defined here. The standardized color set is based on the [X11 color names](https://en.wikipedia.org/wiki/X11_color_names).

Additional resource;
https://www.w3.org/TR/css-color-3/#ref-X11COLORS

### Steps to reproduce

[gd.tumeo.space playground example](https://gd.tumeo.space/KYDwLsB2AmDOAEA5A9tYAodAzArpAxvAPoBOwAhtAJ4AUAlAFzoCQAbuSfIQwLzwDCyADbISAOgBGJZAHdIYsMiIALMAFshNLOSGxgdeAGJ4AWgB88cgDYATAE57LYwGUw5GB2jwAGgEZfAPQAEgAqALIAMvAAQtJypiaJScmJFuQArDbkWabwJADmEjS+VukANPAALDYV1XQsAA4kAJaQYDT49cyG1vb2QA)

### Minimal reproduction project

[gd.tumeo.space playground example](https://gd.tumeo.space/KYDwLsB2AmDOAEA5A9tYAodAzArpAxvAPoBOwAhtAJ4AUAlAFzoCQAbuSfIQwLzwDCyADbISAOgBGJZAHdIYsMiIALMAFshNLOSGxgdeAGJ4AWgB88cgDYATAE57LYwGUw5GB2jwAGgEZfAPQAEgAqALIAMvAAQtJypiaJScmJFuQArDbkWabwJADmEjS+VukANPAALDYV1XQsAA4kAJaQYDT49cyG1vb2QA)

### PR DESCRIPTION
**Pull request by taigi100**
It looks like the BROWN color was a rounding error, using 3 decimals seems to fix it.

- Fixes #59215 

---

As per comments, I've updated most of the values to HEX codes. The only ones remain are `Web` colors, which don't seem to be part of the X11 standard.
@bruvzg

### CODE REVIEW

File: core/math/color_names.inc [hunk 1]
```diff
@@ -24,7 +24,7 @@ static NamedColor named_colors[] = {
 	{ "BLANCHED_ALMOND", Color(1.00, 0.92, 0.80) },
 	{ "BLUE", Color(0.00, 0.00, 1.00) },
 	{ "BLUE_VIOLET", Color(0.54, 0.17, 0.89) },
-	{ "BROWN", Color(0.65, 0.16, 0.16) },
+	{ "BROWN", Color(0.647, 0.164, 0.164) },
```
Comment (bruvzg) [inline core/math/color_names.inc:27]: Maybe it's better to define these colors with RGBA hex values from the standard, like:

```suggestion
	{ "BROWN", Color::hex(0xA52A2AFF) },
```

Comment (taigi100) [reply core/math/color_names.inc:27]: I can transform them all if that's what we want. We can also use the Color("a52a2a") format as well.
Also X11 supports both `DIM GRAY` and `DIM GREY` (same color, two names) and we only support `dim gray`, do we want the other one?

File: core/math/color_names.inc [hunk 2]
```diff
@@ -13,151 +13,151 @@ struct NamedColor {
...
+	{ "FIREBRICK", Color::hex(0xB22222FF) },
+	{ "FLORAL_WHITE", Color::hex(0xFFFAF0FF) },
+	{ "FOREST_GREEN", Color::hex(0x228B22FF) },
+	{ "FUCHSIA", Color::hex(0xFF00FFFF) },
+	{ "GAINSBORO", Color::hex(0xDCDCDCFF) },
+	{ "GHOST_WHITE", Color::hex(0xF8F8FFFF) },
+	{ "GOLD", Color::hex(0xFFD700FF) },
+	{ "GOLDENROD", Color::hex(0xDAA520FF) },
+	{ "GRAY", Color::hex(0x808080FF) },
+	{ "GREEN", Color::hex(0x008000FF) },
```
Comment (bruvzg) [inline core/math/color_names.inc:67]: 0x008000FF is WEB_GREEN, this one should be:

```suggestion
	{ "GREEN", Color::hex(0x00FF00FF) },
```

**Comment by bruvzg** [comment]
Tests should be updated as well:
https://github.com/godotengine/godot/blob/28774e652e4d656bdeec7bb0c26f7451297b5cb0/tests/core/math/test_color.h#L153-L158

File: core/math/color_names.inc [hunk 3]
```diff
@@ -13,151 +13,151 @@ struct NamedColor {
...
+	{ "DODGER_BLUE", Color::hex(0x1E90FFFF) },
+	{ "FIREBRICK", Color::hex(0xB22222FF) },
+	{ "FLORAL_WHITE", Color::hex(0xFFFAF0FF) },
+	{ "FOREST_GREEN", Color::hex(0x228B22FF) },
+	{ "FUCHSIA", Color::hex(0xFF00FFFF) },
+	{ "GAINSBORO", Color::hex(0xDCDCDCFF) },
+	{ "GHOST_WHITE", Color::hex(0xF8F8FFFF) },
+	{ "GOLD", Color::hex(0xFFD700FF) },
+	{ "GOLDENROD", Color::hex(0xDAA520FF) },
+	{ "GRAY", Color::hex(0x808080FF) },
```
Comment (bruvzg) [inline core/math/color_names.inc:66]: Same here, should be `BEBEBEFF`, that's why we have "WEB" variants (these are different between X11 and WEB).

File: core/math/color_names.inc [hunk 4]
```diff
@@ -13,151 +13,151 @@ struct NamedColor {
...
+	{ "LIGHT_SEA_GREEN", Color::hex(0x20B2AAFF) },
+	{ "LIGHT_SKY_BLUE", Color::hex(0x87CEFAFF) },
+	{ "LIGHT_SLATE_GRAY", Color::hex(0x778899FF) },
+	{ "LIGHT_STEEL_BLUE", Color::hex(0xB0C4DEFF) },
+	{ "LIGHT_YELLOW", Color::hex(0xFFFFE0FF) },
+	{ "LIME", Color::hex(0x00FF00FF) },
+	{ "LIME_GREEN", Color::hex(0x32CD32FF) },
+	{ "LINEN", Color::hex(0xFAF0E6FF) },
+	{ "MAGENTA", Color::hex(0xFF00FFFF) },
+	{ "MAROON", Color::hex(0x800000FF) },
```
Comment (bruvzg) [inline core/math/color_names.inc:96]: #B03060

File: core/math/color_names.inc [hunk 5]
```diff
@@ -13,151 +13,151 @@ struct NamedColor {
...
+	{ "PALE_GREEN", Color::hex(0x98FB98FF) },
+	{ "PALE_TURQUOISE", Color::hex(0xAFEEEEFF) },
+	{ "PALE_VIOLET_RED", Color::hex(0xDB7093FF) },
+	{ "PAPAYA_WHIP", Color::hex(0xFFEFD5FF) },
+	{ "PEACH_PUFF", Color::hex(0xFFDAB9FF) },
+	{ "PERU", Color::hex(0xCD853FFF) },
+	{ "PINK", Color::hex(0xFFC0CBFF) },
+	{ "PLUM", Color::hex(0xDDA0DDFF) },
+	{ "POWDER_BLUE", Color::hex(0xB0E0E6FF) },
+	{ "PURPLE", Color::hex(0x800080FF) },
```
Comment (bruvzg) [inline core/math/color_names.inc:128]: #A020F0

**Review by bruvzg — APPROVED** [review]
Seems fine (I have not checked every single color).

**Comment by taigi100** [comment]
> Seems fine (I have not checked every single color).

I copy pasted and edited them directly from the X11 docs, should mostly be fine. Didn't check every color either tho.

File: core/math/color_names.inc [hunk 6]
```diff
@@ -13,151 +13,151 @@ struct NamedColor {
...
+	{ "PALE_TURQUOISE", Color::hex(0xAFEEEEFF) },
+	{ "PALE_VIOLET_RED", Color::hex(0xDB7093FF) },
+	{ "PAPAYA_WHIP", Color::hex(0xFFEFD5FF) },
+	{ "PEACH_PUFF", Color::hex(0xFFDAB9FF) },
+	{ "PERU", Color::hex(0xCD853FFF) },
+	{ "PINK", Color::hex(0xFFC0CBFF) },
+	{ "PLUM", Color::hex(0xDDA0DDFF) },
+	{ "POWDER_BLUE", Color::hex(0xB0E0E6FF) },
+	{ "PURPLE", Color::hex(0xA020F0FF) },
+	{ "REBECCA_PURPLE", Color::hex(0xFF0000FF) },
```
Comment (lnicola) [inline core/math/color_names.inc:129]:
```suggestion
	{ "REBECCA_PURPLE", Color::hex(0x663399FF) },
```

**Review by akien-mga — APPROVED** [review]
Compared the whole list to https://en.m.wikipedia.org/wiki/X11_color_names, seems good.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**